relationships.md

```diff
@@ -94,6 +94,23 @@ The 2 items are included as HAL links but are not embedded
 
 An optional leftwardValue and rightwardValue property can be present. It's omitted when it's null.
 
+## Search methods
+
+### Relationship involving specified items
+**/api/core/relationshiptypes/search/byItemsAndType?typeId=<:relationship-type-id>&leftItem=<item-uuid1>[&...&&leftItem=<item-uuidN>]&rightItem=<item-uuid1>[&...&&rightItem=<item-uuidN>]**
+
+This method is intended to be used when giving an item (focus) and a list of potentially related items we need to know which of these other items are already in a specific relationship with the focus item and, by exclusion which ones are not yet related.
+
+The supported parameters are:
+* page, size [see pagination](README.md#Pagination)
+* typeId: mandatory, the relationship type id to apply as a filter to the returned relationships
+* leftItem; mandatory, repeatable if the rightItem parameter is not repeated. The uuid of the items to be checked on the left side of returned relationships
+* rightItem; mandatory, repeatable if the leftItem parameter is not repeated. The uuid of the items to be found on the right side of returned relationships
```

Comment:
I'm finding the description of this endpoint & the `leftItem` and `rightItem` parameters very confusing.... so I'm not sure I understand how it is supposed to be used

* The description talks about an item that is in "focus".  Which item is the "focus" item, the `leftItem` or `rightItem` one? How do you know which parameter to use here? Do you need to somehow first know whether the focus item is on the left or right side of the given `typeId`?
* I'm I correct in understanding that the items to *check* are then the ones passed into the other param? (i.e. if the `leftItem` is the focus item, then the `rightItem` is the items to check?)

Overall, this endpoint seems confusing to use...and I think it needs clarifying.  I don't quite understand what it should return for specific values....so, maybe examples could help here (even sample `curl` commands with an explanation of what would be returned).   

We also might want to consider whether the params need to be named `leftItem` and `rightItem`...as I worry that adds to the confusion of how to use this endpoint.  Could they be renamed to something like `focusItem` (non-repeatable) and `relatedItem` (repeatable) params?

Reply:
in my intention leftItem and rightItem can be both the "focus".  I assumed that the focus one is the parameter with a single value where the other will have more than one but it really doesn't matter for the server, it is a simple search over the relationship where left and right item are exactly specified.
Who need to worry about the "focus" is the client (angular). Looking to the relationship that it want to manage (i.e. are we talking about the "publications" add button or the "orgunits" add button in a person entity) will put the person uuid in the left or in the right parameter. Please note that we already have and we need a separate "add button" for each relationship that involve our entity and we could have two separate buttons in the case a relationship allow the same entity type on both side (parents - children).

That said, we can also use `focusItem` and `relatedItem` as by your suggestion and I agree that would make the documentation easier but in such case we need to add an additional parameter, other than the typeId, to allow the server to understand which is the side where the focusItem live. I suggest to use label of the relationship, so if the param value is "isAuthorOfPublication" this would mean that the focusItem is on the left, if it is "isPublicationOfAuthor" the focusItem is on the right. Please note that we will still need the typeId as we cannot exclude that the same "label" is reused across different relationship types.

Reply:
@abollini : So, the intention of this endpoint is to *limit* the results by a particular side of the relationship? If that's so, then the side of relationship seems like it should be a parameter in some manner...it'd make this endpoint a lot clearer.

Is there ever a use case where we'd want to NOT limit by the side of the relationship?  E.g. find all Items that are related to my `focusItem`, regardless of what side of the relationship they are on.

If so, that's even more of an argument for making the side of the relationship part of the params.

I'm OK with adding the relationshipLabel as an extra param.  It might be **optional** if we wanted to support finding all related items regardless of relationship.

Reply:
we have not yet such use case and I'm not able to figure out one right now. We can always remove the mandatory flag on the parameter at a later stage without breaking compatibility so I suggest to start simpler and implement it only if and when needed.

Reply:
@abollini : Maybe I'm oversimplifying things, but couldn't we just do this...

Rename `focusSide` to be `relationshipLabel`.   Update the description to say that this endpoint is returning only `relatedItems` which are related to the `focusItem` via the specified `relationshipLabel`.    

So, an example might then be...
`/api/core/relationships/search/byItemsAndType?typeId=<:type-id>&relationshipLable=isAuthorOfPublication&focusItem=<:publication-uuid>&relatedItem=<:one-person-uuid>&relatedItem=<:two-person-uuid>&relatedItem=<:three-person-uuid>`

With that example, the resulting response should be all the `isAuthorOfPublication` relationships where the `<:publication-uuid>` is linked to one of those three Person UUIDs.  (Sidenote, it might even be possible to drop the `typeId` param, but I'm OK with it existing if it needs to.)

My point in this renaming is that I don't think we need to get into details about "leftType" and "leftwardType", etc. in defining this endpoint.

Beyond that, your updated version looks good to me.  We may just want to include an example, like I've included in this comment.

Reply:
@tdonohue I have updated the contract, I will appreciate if you can take a second look to the wording and suggest further improvements if needed